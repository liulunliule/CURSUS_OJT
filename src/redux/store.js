import { combineReducers, configureStore } from "@reduxjs/toolkit";
import savedCourseReducer from "./features/savedCourseSlice";
import blogReducer from './features/ourBlogSlice';
import officesReducer from './features/officesSlice';
import pressReducer from './features/pressSlice';
import payoutReducer from './features/payoutSlice';
import creditsReducer from "./features/creditsSlice";
import certificateReducer from "./features/myCertificateSlice";
import myProfileReducer from "./features/myProfileSlice";
import myHeaderReducer from "./features/myHeaderSlice";
import mySearchReducer from "./features/mySearchSlice";
import myStudioReducer from "./features/myStudioSlice";
import allInstructorReducer from "./features/allInstructorSlice";
import reviewFeedbackReducer from "./features/reviewfeedbackSlice";
import feedbackReportReducer from "./features/feedbackvsReportSlice";
import notificationsSettingReducer from "./features/NotificationsSettingSlice";
import privacySettingReducer from "./features/PrivacySettingSlice";
import userSlice from "./features/userSlice";
import certificationTestPageReducer from "./features/certificationTestPageSlice";
import certificationTestResultReducer from "./features/certificationTestResultSlice";
import analyticsReducer from "./features/analyticsSlice";
import storage from "redux-persist/lib/storage";
import { persistReducer, persistStore } from "redux-persist";
import courseSlice from "./features/courseSlice";
import checkoutReducer from './features/checkoutSlice';
import earningReducer from './features/earningSlice';
import statementReducer from './features/statementSlice';
import shoppingCartReducer from './features/shoppingCartSlice';

const persistConfig = {
    key: "root",
    storage,
    whitelist: [
        "user",
        "savedCourse",
        "certificate",
        "course",
        "myProfile",
        "myHeader",
        "mySearch",
        "myStudio",
        "allInstructor",
        "reviewFeedback",
        "feedbackReport",
        "notificationsSetting",
        "privacySetting",
    ],
};

const rootReducer = combineReducers({
    user: userSlice,
    savedCourse: savedCourseReducer,
    certificate: certificateReducer,
    certificationTestPage: certificationTestPageReducer,
    certificationTestResult: certificationTestResultReducer,
    analytics: analyticsReducer,
    course: courseSlice,
    myProfile: myProfileReducer,
    myHeader: myHeaderReducer,
    mySearch: mySearchReducer,
    myStudio: myStudioReducer,
    allInstructor: allInstructorReducer,
    reviewFeedback: reviewFeedbackReducer,
    feedbackReport: feedbackReportReducer,
    checkout: checkoutReducer,
    earnings: earningReducer,
    statement: statementReducer,
    credits: creditsReducer,
    payout: payoutReducer,
    blogs: blogReducer,
    offices: officesReducer,
    press: pressReducer,
    notificationsSetting: notificationsSettingReducer,
    privacySetting: privacySettingReducer,
    shoppingCart: shoppingCartReducer,
});

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = configureStore({
    reducer: persistedReducer,
});

export const persistor = persistStore(store);
