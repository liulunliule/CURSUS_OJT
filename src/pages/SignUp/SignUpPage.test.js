import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom'; // Import MemoryRouter
import { store } from '../../redux/store';
import SignUpPage from '.';

test('test SignUpPage Page', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <SignUpPage/>
      </MemoryRouter>
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
