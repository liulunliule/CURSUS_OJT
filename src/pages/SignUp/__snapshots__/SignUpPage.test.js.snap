// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test SignUpPage Page 1`] = `
<div>
  <div
    className="signup"
  >
    <img
      alt=""
      className="signup__logo"
      src="logoCursus.svg"
    />
    <div
      className="signup__main main"
    >
      <h2
        className="signup__title"
      >
        Welcome to Cursus
      </h2>
      <p
        className="signup__desc"
      >
        Sign Up and Start Learning!
      </p>
      <div
        className="signup__cta"
      >
        <div
          action=""
          className="signup__form-group"
        >
          <div
            className="signup__form"
          >
            <input
              className="signup__form-input"
              id=""
              name=""
              onChange={[Function]}
              placeholder="User Name"
              type="text"
              value=""
            />
          </div>
          <div
            className="signup__form"
          >
            <input
              className="signup__form-input"
              id=""
              name=""
              onChange={[Function]}
              placeholder="Email Address"
              type="text"
              value=""
            />
          </div>
          <div
            className="signup__form"
          >
            <input
              className="signup__form-input"
              id=""
              name=""
              onChange={[Function]}
              placeholder="Password"
              type="text"
              value=""
            />
          </div>
          <div
            className="signup__remember"
          >
            <input
              id="remember-me"
              name=""
              type="checkbox"
            />
            <label
              for="remember-me"
            >
              I’m in for emails with exciting discounts and personalized recommendations
            </label>
          </div>
          <button
            className="signup__button signup__signin"
            onClick={[Function]}
          >
            Next
          </button>
        </div>
      </div>
      <div
        className="signup__forgotPass"
      >
        By signing up, you agree to our
         
        <a
          href="/secondLayout/policy"
          onClick={[Function]}
        >
           Terms of Use
        </a>
         and
         
        <a
          href="/secondLayout/term"
          onClick={[Function]}
        >
           Privacy Policy
        </a>
        .
      </div>
      <div
        className="signup__line"
      />
      <div
        className="signup__signup"
      >
        Already have an account?
        <a
          href="/login"
          onClick={[Function]}
        >
          Log in
        </a>
      </div>
    </div>
    <div
      className="signup__footer"
    >
      <img
        alt=""
        src="small_logo.png"
      />
      <p
        className="login__footer-content"
      >
        © 2024 
        <b>
          Cursus
        </b>
        . All Rights Reserved.
      </p>
    </div>
  </div>
</div>
`;
